config/routes.rb

```diff
@@ -817,6 +808,40 @@ def route_actions_hash
   get("/observer/show_rss_log/:id", to: redirect("/activity_logs/%{id}"))
   get("/observer/rss", to: redirect("/activity_logs/rss"))
 
+  # ----- Sequences: legacy action redirects
+  redirect_legacy_actions(
+    old_controller: "sequence", new_controller: "sequences",
+    # actions: [:show, :list, :index]
+    actions: [:show, :index]
+  )
+  # ----- Sequences: nonstandard legacy action redirects
+  # get("/sequence/index", to: redirect("/sequences"))
+  # get("/sequence/index_sequence", to: redirect("/sequences"))
+  get("/sequence/list_sequences", to: redirect("/sequences?flavor=all"))
+  get("/sequence/observation_index",
+      # Disable cops; Rails doesn't allow do/end syntax in redirect blocks
+      # https://api.rubyonrails.org/classes/ActionDispatch/Routing/Redirection.html#method-i-redirect
+      to: redirect { |_params, request| # rubocop:disable Style/BlockDelimiters
+            # Does original request have params?
+            URI.parse(request.original_url).query ? # rubocop:disable Style/MultilineTernaryOperator
+              # preserve original id param (and any others)
+              "/sequences?#{URI.parse(request.original_url).query}" \
+              "&flavor=observation"
+            :
+              "/sequences"
+          })
+  get("/sequence/sequence_search",
+      # Disable cops; Rails doesn't allow do/end syntax in redirect blocks
+      # https://api.rubyonrails.org/classes/ActionDispatch/Routing/Redirection.html#method-i-redirect
+      to: redirect { |_params, request| # rubocop:disable Style/BlockDelimiters
+            # Does original request have params?
+            URI.parse(request.original_url).query ? # rubocop:disable Style/MultilineTernaryOperator
+              # preserve original pattern param (and any others)
+              "/sequences?#{URI.parse(request.original_url).query}"
+            :
+              "/sequences"
+          })
+
```

Comment:
Yikes!  Is there any better way to do this?  Except for the specialized /id parameter, don't all the ?params come along for the ride automatically with redirects?  I do remember having to explicitly route /observer/blah/id and /observer/blah separately, but I don't remember seeing all this complex parsing in the legacy observer controller actions.  Sorry, I should look into this more myself, but maybe you've already been there and done that?

Reply:
Bottom lines: 
1. It doesn't matter here as this redirect route will be trashed. See discussion above https://github.com/MushroomObserver/mushroom-observer/pull/1124#discussion_r947067090
2. I think you're right for all practical purposes. And I was confused
For instance, the legacy URL
```http
https://mushroomobserver.org/observer/pattern_search?pattern=date%3A2015-11-20+user%3A%22kundabungkid%22+region%3AAustralia
```
lands on the right page (with a weird flash message `Search is invalid: ‘nil’`), with the address bar showing
```http
https://mushroomobserver.org/observations?pattern=date%3A2015-11-20+user%3A%22kundabungkid%22+region%3AAustralia
```
I got confused by web posts like
this: https://relativkreativ.at/articles/how-to-preserve-the-query-string-when-redirecting-routes-in-a-rails-application
and this: https://www.viget.com/articles/merging-query-strings-when-redirecting-in-rails/
and this: https://stackoverflow.com/questions/19549203/rails-routes-redirect-for-query-string